src/box/relay.cc

```diff
@@ -482,8 +482,9 @@ relay_initial_join(struct iostream *io, uint64_t sync, struct vclock *vclock,
 	 * All these versions know of additional META stage of initial join.
 	 */
 	ctx.send_meta = replica_version_id > 0;
-	ctx.vclock = vclock;
 	ctx.cursor = cursor;
+	/* If cursor is passed, we should respond with its vclock. */
+	ctx.vclock = cursor != NULL ? cursor->vclock : vclock;
```

Comment:
I see what. But why?

Reply:
Elaborated here:
```c
/*
 * If cursor is passed, we should respond with its vclock because
 * it will actually be vclock of a last sent row.
 */
```